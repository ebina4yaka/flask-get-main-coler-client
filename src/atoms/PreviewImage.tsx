import { ReactElement } from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { createStyles } from '@material-ui/core'

type Props = {
  image: string
}

const useStyles = makeStyles(
  createStyles({
    root: {
      maxWidth: '80%',
      maxHeight: '80%',
    },
    message: {
      fontSize: '1.5em',
    },
  })
)

export default function PreviewImage(props: Props): ReactElement {
  const { image } = props
  const classes = useStyles()
  if (image) {
    return <img className={classes.root} alt="preview" src={image} />
  }
  return <div className={classes.message}>Please select image file</div>
}
