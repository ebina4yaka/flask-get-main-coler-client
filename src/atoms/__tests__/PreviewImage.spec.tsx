import renderer from 'react-test-renderer'
import PreviewImage from '../PreviewImage'

test('Snapshot test for PreviewImage', () => {
  const tree = renderer.create(<PreviewImage image="test" />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot test for PreviewImage empty', () => {
  const tree = renderer.create(<PreviewImage image="" />).toJSON()
  expect(tree).toMatchSnapshot()
})
