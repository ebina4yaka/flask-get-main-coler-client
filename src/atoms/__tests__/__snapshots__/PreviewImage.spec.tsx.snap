// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for PreviewImage 1`] = `
<img
  alt="preview"
  className="makeStyles-root-1"
  src="test"
/>
`;

exports[`Snapshot test for PreviewImage empty 1`] = `
<div
  className="makeStyles-message-2"
>
  Please select image file
</div>
`;
